src/VisualStudio/Core/Def/LanguageService/AbstractLanguageService`2.VsLanguageDebugInfo.cs

```diff
@@ -71,161 +71,146 @@ public int GetLocationOfName(string pszName, out string? pbstrMkDoc, out VsTextS
+                // NOTE(cyrusn): cLines is ignored.  This is to match existing dev10 behavior.
+                using (Logger.LogBlock(FunctionId.Debugging_VsLanguageDebugInfo_GetProximityExpressions, CancellationToken.None))
                 {
-                    _uiThreadOperationExecutor.Execute(
+                    using var context = _uiThreadOperationExecutor.BeginExecute(
                         title: ServicesVSResources.Debugger,
                         defaultDescription: ServicesVSResources.Determining_autos,
                         allowCancellation: true,
-                        showProgress: false,
-                        action: context =>
-                    {
-                        var textBuffer = _languageService.EditorAdaptersFactoryService.GetDataBuffer(pBuffer);
+                        showProgress: false);
 
-                        if (textBuffer != null)
-                        {
-                            var snapshot = textBuffer.CurrentSnapshot;
-                            var nullablePoint = snapshot.TryGetPoint(iLine, iCol);
-                            if (nullablePoint.HasValue)
-                            {
-                                var document = snapshot.GetOpenDocumentInCurrentContextWithChanges();
-                                if (document != null)
-                                {
-                                    var point = nullablePoint.Value;
-                                    var proximityExpressions = _proximityExpressionsService.GetProximityExpressionsAsync(document, point.Position, context.UserCancellationToken).WaitAndGetResult(context.UserCancellationToken);
-
-                                    if (proximityExpressions != null)
-                                    {
-                                        enumBSTR = new VsEnumBSTR(proximityExpressions);
-                                    }
-                                }
-                            }
-                        }
-                    });
+                    if (_proximityExpressionsService == null)
+                        return null;
+
+                    var textBuffer = _languageService.EditorAdaptersFactoryService.GetDataBuffer(pBuffer);
+                    if (textBuffer == null)
+                        return null;
+
+                    var snapshot = textBuffer.CurrentSnapshot;
+                    var nullablePoint = snapshot.TryGetPoint(iLine, iCol);
+                    if (!nullablePoint.HasValue)
+                        return null;
+
+                    var document = snapshot.GetOpenDocumentInCurrentContextWithChanges();
+                    if (document == null)
+                        return null;
+
+                    var point = nullablePoint.Value;
+                    var proximityExpressions = await _proximityExpressionsService.GetProximityExpressionsAsync(
+                        document, point.Position, context.UserCancellationToken).ConfigureAwait(true);
```

Comment:
There's places you're doing a ConfigureAwait(true) before we return a VsEnumBSTR and also places we you are donig ConfigureAwait(false); for example ResolveName below. I don't imagine we need to be on the UI thread at that point so I think it's fine, but I'd ask that either we make them consistently ConfigureAwait(true) or ConfigureAwait(false).

Reply:
Platform guidance is anything known to only be called from a jtf run should be CA(true). That way the main thread can execute those callbacks instead of blocking waiting for a big thread to do it.